test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -319,3 +329,203 @@
                           :fields [[:field (mt/id :times :index)]]}}]]
         (testing title
           (is (= (set expected) (set (test-datetime-math query)))))))))
+
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                           Convert Timezone tests                                               |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+(def offset->zone
+  "A map of all Offset to a zone-id.
+  {\"+07\00\" \"Asia/Saigon\"}"
+  (into {"+00:00" "UTC"}
+        (for [zone-id (java.time.ZoneId/getAvailableZoneIds)]
+          [(-> (t/zone-id zone-id)
+               .getRules
+               (.getOffset (java.time.Instant/now))
+               .toString)
+           zone-id])))
```

Comment:
A function from offset to zone-id doesn't make sense as a function, since a time zone can correspond to any number of offsets depending on the point in time (because of daylight savings and changes in time zone laws). That's why `.getOffset` takes a date as an argument. This function needs to take a date as an argument and pass it to `.getOffset` in order to make sense.

Reply:
yea you're right. even though most tests are run with `Asia/Tokyo` and `Asia/Ho_Chi_Minh` in which these places are not affected by DST, it's still confusing to use this function.

I'll remove it and use the full timezone-id in tests.

Reply:
4e5c9ca4fdbd60f616a0a0e5a5341945cb91bd42